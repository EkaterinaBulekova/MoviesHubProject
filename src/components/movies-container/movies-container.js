import React from 'react';
import MovieList from '../movie-list/movie-list';
import Message from '../message/message';

export class MoviesContainer extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      loaded: false,
      movies: [],
    };
  }

  componentDidMount(){
    (this.props.filter.search)
    ? this.props.getData(this.props.filter)
        .then(result => this.setState({loaded: true, movies: result.data}))
        .then(()=>this.props.onUpdate(this.state.movies.length))
    : this.setState({loaded: false, movies: []});
  }

  shouldComponentUpdate(nextProps, nextState) {
    if (nextProps.filter.search !== this.props.filter.search ||
      nextProps.filter.searchBy !== this.props.filter.searchBy ||
      nextProps.filter.sortBy !== this.props.filter.sortBy){
    (nextProps.filter.search)
      ? this.props.getData(nextProps.filter)
          .then(result => this.setState({loaded: true, movies: result.data}))
          .then(()=>this.props.onUpdate(this.state.movies.length))
      : this.setState({loaded: false, movies: []});
      return false;
   }
   return (nextState.loaded && 
            (this.state.movies.length !== nextState.movies.length ||
             nextState.movies.length && nextState.movies.filter((movie,index)=>movie.id !== this.state.movies[index].id).length))
          || !nextState.movies.length;
  }

  render() {
      let selectedMovie = this.props.filter.selectedMovie
      let movies = (selectedMovie)
        ? this.state.movies.filter(movie=>movie.id !== selectedMovie.id)
        : this.state.movies;
    return (
      movies.length
        ? <MovieList movies={movies} onClick={(movie) => this.props.onClick(movie)} />
        :this.state.loaded && <Message text="No films found"></Message>
    )
  }
}