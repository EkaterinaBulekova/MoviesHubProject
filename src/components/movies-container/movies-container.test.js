import React from 'react';
import { mount } from 'enzyme';
import { MoviesContainer } from './movies-container';
import toJson from 'enzyme-to-json';

describe('<MoviesContainer />', () => {
  it('should render list of movie card with right title, class and metod', () => {
    const getTestData = jest.fn(() => Promise.resolve({data:[
      {
          id: 1,
          title: "Test movie1",
          poster_path: "http://test-path",
          release_date:"2019-01-05",
          genres: "test genre1"
      },
      {
          id: 2,
          title: "Test movie2",
          poster_path: "http://test-path",
          release_date:"2019-01-05",
          genres: "test genre2"
      }]}));
    const testFilter = {search: 'value', searchBy: 'title', sortBy: 'rating'};

    const testOnClick = jest.fn();
    const testOnUpdate = jest.fn();
    const component = mount(<MoviesContainer filter={testFilter} onClick={testOnClick} getData={getTestData} onUpdate={testOnUpdate}/>);

    expect(toJson(component)).toMatchSnapshot();
    expect(getTestData).toHaveBeenCalledTimes(1);
    expect(getTestData.mock.calls.length).toBe(1);
    expect(getTestData).toBeCalled();
    expect(getTestData).toBeCalledWith(testFilter);
    component.update();
    expect(component.s)

  })

});