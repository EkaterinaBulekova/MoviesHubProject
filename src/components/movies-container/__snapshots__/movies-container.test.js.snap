// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MoviesContainer /> should render list of movie card with right title, class and metod 1`] = `
<MoviesContainer
  filter={
    Object {
      "search": "value",
      "searchBy": "title",
      "sortBy": "rating",
    }
  }
  getData={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "search": "value",
            "searchBy": "title",
            "sortBy": "rating",
          },
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": Promise {},
        },
      ],
    }
  }
  onClick={[MockFunction]}
  onUpdate={[MockFunction]}
/>
`;
